docs/core/compatibility/core-libraries/6.0/filestream-file-locks-unix.md

```diff
@@ -0,0 +1,50 @@
+---
+title: "Breaking change: FileStream locks files with shared lock on Unix"
+description: Learn about the .NET 6 breaking change where FileStream.Lock now locks files with a read lock when they're opened with read permissions on Unix.
+ms.date: 10/15/2021
+---
+# FileStream locks files with shared lock on Unix
+
+On Unix, if you open a file using <xref:System.IO.FileStream> with <xref:System.IO.FileAccess.Read?displayProperty=nameWithType> permissions only, and then call <xref:System.IO.FileStream.Lock(System.Int64,System.Int64)?displayProperty=nameWithType> to lock a region of the file, the operation will now succeed. It succeeds because the runtime locks the file with a shared or read lock instead of a write lock.
+
+```csharp
+using (FileStream fs = File.OpenRead("testfile")) // Opening with FileAccess.Read only
+{
+    fs.Lock((long) 3, (long) 1); // Attempting to lock a region of the read-only file
+}
+```
+
+There's no change to the behavior on Windows, where the operation always succeeded.
+
+## Previous behavior
+
+On Unix, if you opened a file with <xref:System.IO.FileStream> with read permissions only, and then called <xref:System.IO.FileStream.Lock(System.Int64,System.Int64)?displayProperty=nameWithType> to lock a region of the file, the runtime attempted to lock the file with a write lock. That resulted in an <xref:System.UnauthorizedAccessException> and the message "Access to the path is denied".
```

Comment:
The two consecutive `with` words tripped me a bit. The description in the next section is easier to read:
 
```suggestion
On Unix, if you opened a file using a <xref:System.IO.FileStream> with read permissions only, and then called <xref:System.IO.FileStream.Lock(System.Int64,System.Int64)?displayProperty=nameWithType> to lock a region of the file, the runtime attempted to lock the file with a write lock. That resulted in an <xref:System.UnauthorizedAccessException> and the message "Access to the path is denied".
```